src/Compilers/Core/Portable/Operations/OperationInterfaces.xml

```diff
@@ -485,6 +485,11 @@
         </summary>
       </Comments>
     </Property>
+    <Property Name="DisposeMethod" Type="IMethodSymbol?">
```

Comment:
><Property Name="DisposeMethod" Type="IMethodSymbol?"> [](start = 4, length = 53)

With this design choice we won't be able to represent different methods for different resources. This is probably sufficient today because only C# supports pattern based disposing and it cannot have resources of different types. However, VB can have resources of different types and, if we decide to support pattern based disposing in VB, we will have a hard time representing that in the API. Consider accounting for that in the shape of the API. For example, instead of a property, we could have a method that returns a MethodSymbol for specific resource identified either by index, or by specific node, representing the resource (IVariableDeclarationOperation or the singleton).

Reply:
Or the property can return an array of method symbols, but that would force us to always allocate an array which today will always be filled with the same value.   

---
In reply to: [528014394](https://github.com/dotnet/roslyn/pull/49481#discussion_r528014394) [](ancestors = 528014394)

Reply:
That's interesting. I guess I figured we were safe from a VB point of view because we've said we wouldn't evolve it further. Agree about the array case. 

How would you envisage the method API working? That we'd have a method on the operation itself that returns the dispose method for each possible child resource?

Reply:
> How would you envisage the method API working? That we'd have a method on the operation itself that returns the dispose method for each possible child resource? 

Yes, something like that.

---
In reply to: [528022713](https://github.com/dotnet/roslyn/pull/49481#discussion_r528022713) [](ancestors = 528022713)

Reply:
@AlekseyTs Hmm, I started playing around with this, and I think it's probably do-able, but requires a bit of re-layering. Given that we didn't previously expose these publically, would it be acceptable to just make the `DisposeMethod` `internal` for this PR, and file a an issue / follow up PR to expose it via a suitable public API?

Reply:
> Given that we didn't previously expose these publically, would it be acceptable to just make the DisposeMethod internal for this PR, and file a an issue / follow up PR to expose it via a suitable public API?

I think that would be the right thing to do at the moment. We are not exposing this information publicly for foreach, for example. 

---
In reply to: [528029327](https://github.com/dotnet/roslyn/pull/49481#discussion_r528029327) [](ancestors = 528029327)